canonicalization.md

```diff
@@ -0,0 +1,19 @@
+# Canonicalization
+
+One benefit of content-addressable storage is easy deduplication.
+Many images might depend on a particular [layer](layer.md), the there will only be one blob in the [store](image-layout.md).
+However, that same semantic layer serialized slightly differently would have a different hash, and if both versions of the layer are referenced there will be two blobs with the same semantic content.
+To allow efficient storage, implementations serializing content for blobs SHOULD use a canonical serialization.
+This increases the chance that different implementations can push the same semantic content to the store without creating redundant blobs.
+
+## JSON
+
+[JSON][] content SHOULD be serialized as [canonical JSON][canonical-json].
```

Comment:
JSON content (for example all of the bla bla media types or bla bla manifests) SHOULD be serialized as